### ISSUE TITLE
Clickable pins and random values to locations

### ISSUE THREAD

**Issue by egemen-dev** [kasugaijin/baja-pet-rescue#158]
I copied the repo and got it running on my local environment to experiment adding clickable pins and meta data. Here is the **preview**. I might open a PR after this. https://github.com/kasugaijin/baja-pet-rescue/issues/144
![Screenshot 2023-06-11 at 14-11-15 BajaPetRescue](https://github.com/kasugaijin/baja-pet-rescue/assets/93445248/bafa63ea-19c8-4e9a-a3e4-59ab01bebb58)

---

My question is, the reason to apply random values to the duplicate locations (https://github.com/kasugaijin/baja-pet-rescue/issues/134) because if the same person adopted multiple pets it's nearly impossible to individually display them on the map - they overlay each other, is that right or something else? @kasugaijin

**Comment by luuu-xu** [comment]
Hi, just stumbled upon your issue, I would like to guide you towards solving this with Google Maps' [Marker Cluster](https://developers.google.com/maps/documentation/javascript/marker-clustering) feature. I have encountered this same issue on my [risk-viz](https://github.com/luuu-xu/risk-viz) project. Only using marker cluster would not be enough since if you have multiple markers at the exact same location, Google Maps wouldn't show them all. In the end I used marker cluster, with a helper function which adds a little bit of [variance](https://github.com/luuu-xu/risk-viz/blob/main/src/app/lib/addVarianceToData.ts) to the lat,long data so it shows up perfectly. Hope it helps!

**Comment by kasugaijin** [comment]
Hey @egemen-dev thank you for opening this. This would be a great addition to show Dog name (more data could be added if it made sense. I am open to suggestions). 

There's a couple of related issues, ad you have pointed out. They should be done in this sequence:

- https://github.com/kasugaijin/baja-pet-rescue/issues/134 this applies a random adjustment to pins that would otherwise overlap. Yes you're correct, if one person adopted multiple pets, or two people living in the same city adopted, their pins would overlap. I originally approach this problem by identifying the overlapping locations and then applying a deviation to them via the `SuccessesController::GoogleMapsCoordinates`. Because it's a more complex approach, I did not bother to maintain metadata like dog name etc. However, it's excessive and unnecessary to do it this way, and instead I think we can just apply a random deviation to every location, and maintain its metadata for the pin. This class should be refactored accordingly. Once that is done, and once the map starts to look really busy, I think we can open another issue and implement @luuu-xu solution using clusters...but I don't think we need to do that just yet.
- https://github.com/kasugaijin/baja-pet-rescue/issues/144 can be done next
- then this issue you created here

**Comment by kasugaijin** [comment]
This PR now merged: https://github.com/kasugaijin/baja-pet-rescue/pull/160
So this one can continue to add meta data to dog pins. See my comment on the PR. 

Thanks

**Comment by kasugaijin** [comment]
Closed.
